packages/superset-ui-chart/src/query/processFilters.ts

```diff
@@ -0,0 +1,51 @@
+import { ChartFormData } from '../types/ChartFormData';
+import { QueryObjectFilterClause } from '../types/Query';
+import { isSimpleAdhocFilter } from '../types/Filter';
+import convertFilter from './convertFilter';
+
+/** Logic formerly in viz.py's process_query_filters */
+export default function processFilters(formData: ChartFormData) {
+  // TODO: Implement
+  // utils.convert_legacy_filters_into_adhoc(self.form_data)
+
+  // TODO: Implement
```

Comment:
Will these two in the scope of this PR?

Reply:
These will be beyond the scope. Right now this PR will handle `adhoc_filters` correctly plus provide typings.

Reply:
do you think we should `console.warn` if we see legacy filters? (fine if not)

Reply:
The legacy filters are using fields `filters` and `having_filters` which do not exist on type `ChartFormData`, so typescript would complain.

I am wondering if we want to support this `utils.convert_legacy_filters_into_adhoc(self.form_data)` going forward. Not sure how long since we have moved from the legacy filters.